src/components/TagContainer.ts

```diff
@@ -0,0 +1,261 @@
+    class?: string;
+    mxObject: mendix.lib.MxObject;
+    mxform: mxui.lib.form._FormBase;
+    readOnly: boolean;
+    style?: string;
+}
+
+export interface TagContainerProps extends WrapperProps {
+    afterCreateMicroflow: string;
+    allowDrag: boolean;
+    tagStyle: BootstrapStyle;
+    editable: "default" | "never";
+    enableSuggestions: boolean;
+    inputPlaceholder: string;
+    onChangeMicroflow: string;
+    lazyLoad: boolean;
+    tagAttribute: string;
+    tagConstraint: string;
+    tagEntity: string;
+    tagLimit: number;
+    tagLimitMessage: string;
+}
+
+export interface TagState {
+    isReference: boolean;
+    alertMessage?: string;
+    suggestions: string[];
+    lazyLoaded: boolean;
+    tagList: string[];
+    newTag: string;
+    tagCache: mendix.lib.MxObject[];
+}
+
+export default class TagContainer extends Component<TagContainerProps, TagState> {
+    private subscriptionHandles: number[] = [];
+    private tagEntity: string;
+    private referenceAttribute: string;
+
+    constructor(props: TagContainerProps) {
+        super(props);
+
+        this.state = {
+            isReference: false,
+            newTag: "",
+            suggestions: [],
+            lazyLoaded: false,
+            tagCache: [],
+            tagList: []
+        };
+        this.createTag = this.createTag.bind(this);
+        this.processTagData = this.processTagData.bind(this);
+        this.lazyLoadSuggestions = this.lazyLoadSuggestions.bind(this);
+        this.removeTag = this.removeTag.bind(this);
+        this.showAlertMessage = this.showAlertMessage.bind(this);
+
+        this.tagEntity = props.tagEntity.split("/")[props.tagEntity.split("/").length - 1];
+        this.referenceAttribute = props.tagEntity.split("/")[0];
+    }
+
+    render() {
+        return createElement(Tag, {
+            alertMessage: this.state.alertMessage,
+            tagStyle: this.props.tagStyle,
+            className: this.props.class,
+            createTag: this.createTag,
+            enableSuggestions: this.props.enableSuggestions,
+            inputPlaceholder: this.props.inputPlaceholder,
+            lazyLoad: this.props.lazyLoad,
+            fetchSuggestions: this.lazyLoadSuggestions,
+            newTag: this.state.newTag,
+            onRemove: this.removeTag,
+            readOnly: this.isReadOnly(),
+            showError: this.showAlertMessage,
+            style: parseStyle(this.props.style),
+            suggestions: this.state.suggestions,
+            tagLimit: this.props.tagLimit,
+            tagLimitMessage: this.props.tagLimitMessage,
+            tagList: this.state.tagList
+        });
+    }
+
+    componentWillReceiveProps(newProps: TagContainerProps) {
+        this.fetchTagData(newProps.mxObject);
+        this.resetSubscriptions(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+    }
+
+    private showAlertMessage(message: string) {
+
```

Comment:
Remove space since return is the only one

Reply:
Done

Reply:
Done